app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SearchController.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc._
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.forms.{SearchForm, SearchFormData}
+import uk.gov.hmrc.tariffclassificationfrontend.models.Sort.Sort
+import uk.gov.hmrc.tariffclassificationfrontend.models.{Search, Sort}
+import uk.gov.hmrc.tariffclassificationfrontend.service.CasesService
+import uk.gov.hmrc.tariffclassificationfrontend.views.html
+
+import scala.concurrent.Future
+import scala.concurrent.Future.successful
+
+@Singleton
+class SearchController @Inject()(authenticatedAction: AuthenticatedAction,
+                                 casesService: CasesService,
+                                 val messagesApi: MessagesApi,
+                                 implicit val appConfig: AppConfig) extends FrontendController with I18nSupport {
+
+  def search(reference: Option[String] = None, search: Search = Search(), sort: Sort = Sort.COMMODITY_CODE): Action[AnyContent] = authenticatedAction.async { implicit request =>
+    if (reference.isDefined) {
```

Comment:
Does this belong on this endpoint? Seems like a separate concern to searching

Reply:
Yes perhaps, I'll have a think

Reply:
As mentioned on slack, frustratingly play doesnt let us route `/search` to two different controller endpoints based on query params. We would have to have another endpoint e.g. `/search/ref?reference=:ref` but I'd rather keep everything under `/search`